lib/lintPostcssResult.js

```diff
@@ -22,7 +22,7 @@ function lintPostcssResult(stylelintOptions, postcssResult, config) {
 	postcssResult.stylelint.quiet = config.quiet;
 	postcssResult.stylelint.config = config;
 
-	/** @type {string} */
+	/** @type {string | undefined} */
```

Comment:
@ybiquitous `postcssDoc` at l29 is always truthy correct?
I am trying to check if the `newline == null` conditions are necessary.
Isn't `newline` always a string?

Reply:
Um, I don't remember it well. In the current PostCSS type definitions, the `Result.root` property always seems truthy:
- https://github.com/postcss/postcss/blob/8.4.38/lib/result.d.ts#L144
- https://github.com/postcss/postcss/blob/8.4.38/lib/lazy-result.js#L145

But I'm unsure, honestly. If so, we can make this code much simpler, including `newline`... 🤔

Reply:
I think it had consequences elsewhere. Anyway, I am gonna change its type to be `string` in an upcoming refactor.